File: app/src/main/java/me/androidbox/busbymovies/movielist/IMovieListViewHolderFactory.java
package me.androidbox.busbymovies.movielist;

import android.view.View;

import me.androidbox.busbymovies.adapters.MovieAdapter;
import me.androidbox.busbymovies.utils.ImageLoader;

public interface IMovieListViewHolderFactory {
    MovieListViewHolder createViewHolder(final View parent,
                                         final MovieAdapter movieAdapter,
                                         final ImageLoader imageLoader,
                                         final MovieListItemClickedListener movieListItemClickedListener);
}

File: app/src/main/kotlin/me/androidbox/busbymovies/movielist/IMovieListViewHolderFactory.kt
package me.androidbox.busbymovies.movielist

import android.support.v7.widget.RecyclerView
import android.view.View
import me.androidbox.busbymovies.utils.ImageLoader

interface IMovieListViewHolderFactory {
    fun createViewHolder(parent: View, imageLoader: ImageLoader): RecyclerView.ViewHolder
}
